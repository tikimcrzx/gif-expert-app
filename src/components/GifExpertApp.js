import React, { useState } from 'react';
import { AddCategory } from './AddCategory';
import { GifGrid } from './GifGrid';

export default function GifExpertApp({ defaultCategories = [] }) {
  // const [categories, setCategories] = useState(['One Punch']);
  const [categories, setCategories] = useState(defaultCategories);

  return (
    <>
      <h2>GifExpertApp</h2>
      <hr />
      <AddCategory setCategories={setCategories} />
      <ol>
        {categories.map((category) => (
          <GifGrid key={category} category={category} />
        ))}
      </ol>
    </>
  );
}
