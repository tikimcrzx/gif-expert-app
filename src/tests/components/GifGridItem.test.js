import React from 'react';
import '@testing-library/jest-dom';
import { shallow } from 'enzyme';
import { GifGridItem } from '../../components/GifGridItem';

describe('Test Gif Grid Item', () => {
  const title = 'title';
  const url = 'http://localhost:3000';
  const wrapper = shallow(<GifGridItem title={title} url={url} />);

  test('Show GifGridItem <GifGridItem />', () => {
    expect(wrapper).toMatchSnapshot();
  });

  test('Content param <p>', () => {
    const p = wrapper.find('p').text().trim();
    expect(p).toBe(title);
  });

  test('Content url image and alt the image', () => {
    const img = wrapper.find('img');
    expect(img.prop('src')).toBe(url);
    expect(img.prop('alt')).toBe(title);
  });

  test('Required class animate__fadeIn', () => {
    const div = wrapper.find('div');
    const className = div.prop('className');
    const classNameCompare = 'animate__fadeIn';
    expect(className.includes(classNameCompare)).toBe(true);
  });
});
