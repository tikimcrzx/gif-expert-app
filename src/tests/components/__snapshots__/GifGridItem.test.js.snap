// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test Gif Grid Item Show GifGridItem <GifGridItem /> 1`] = `
<div
  className="card animate__animated animate__fadeIn"
>
  <img
    alt="title"
    src="http://localhost:3000"
  />
  <p>
    title
  </p>
</div>
`;
